src/Servers/Kestrel/Kestrel/test/KestrelConfigurationLoaderTests.cs

```diff
@@ -840,6 +840,88 @@ public void ConfigureEndpoint_DoesNotThrowWhen_HttpsConfigIsDeclaredInEndpointDe
         Assert.Null(end1.EndpointConfig.SslProtocols);
     }
 
+    [Theory]
+    [InlineData(true)] // This might be flaky
```

Comment:
Why might this be flaky? If there's a solid chance it is, we should quarantine it before we check it in and file an issue. If it really is reliable, it *should* get unquarantined. Or we should just fix the test not to be flaky.

And in the false case, can we make the timeout faster? Not so fast it would reliably pass if we somehow ignored `reloadOnChange` being false ofc.

Reply:
It might be flaky because we don't know how long the file change event will take to arrive (e.g. because of defender).  I removed all file system access from the CertificatePathWatcherTests, but I do insist on having at least one test that actually interacts with a file watcher.

To me, it doesn't make logical sense to make the negative-case delay shorter than the positive-case timeout, but I could be missing something.

Reply:
I don't see how we can make it less flaky, other than by extending the timeout.  I don't think it makes sense to mock the file system in this test.

I dropped the negative-case delay to 1 second.  Statistically, some runs should still fail if we break it.